types/tx/msgs.go

```diff
@@ -7,6 +7,15 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+const (
+	msgResponseInterfaceProtoName = "cosmos.tx.v1beta1.MsgResponse"
+)
+
+// MsgResponse is the interface all Msg server handlers' response types need to
+// implement. It's the interface that's representing all Msg responses packed
+// in Anys.
+type MsgResponse interface{}
```

Comment:
Why is MsgResponse in tx package, whilst Msg is in sdk package. Maybe both fit inside the tx package? We could move sdk.Msg to tx and create a type alias in sdk/types to keep backwards compat

Reply:
Good idea

Reply:
> We could move sdk.Msg to tx and create a type alias in sdk/types to keep backwards compat

Actually, this creates a sdk & tx dep cycle. would you be okay if I created an issue to track this? Moreover, this PR is getting big, would like to scope it to middleware stuff.

Reply:
https://github.com/cosmos/cosmos-sdk/issues/10650

Reply:
oh i see.. but yeah agreed on keeping the scope small